cmd/keys_rotate.go

```diff
@@ -0,0 +1,360 @@
+	const rotateCmd = `#!/usr/bin/python3
+import datetime
+import json
+import os
+import subprocess
+import sys
+from tempfile import TemporaryDirectory
+
+def cmd(*args, cwd=None):
+    print('running: %s' % ' '.join(args))
+    p = subprocess.Popen(
+        args, cwd=cwd,stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    for line in p.stdout:
+        sys.stdout.buffer.write(line)
+    p.wait()
+    if p.returncode != 0:
+        raise subprocess.CalledProcessError(p.returncode, args)
+
+def find_current_root(repodir):
+    print("finding root key name and key id")
+    with open(os.path.join(repodir, 'roles/unsigned/root.json')) as f:
+        root_role = json.load(f)
+
+    key_ids = root_role['roles']['root']['keyids']
+    assert len(key_ids) == 1, "Unexpected number of root keys"
+    print("current root keyid:", key_ids[0])
+    pubkey = root_role['keys'][key_ids[0]]["keyval"]["public"]
+
+    # now find pubkey:
+    keydir = os.path.join(repodir, 'keys')
+    for x in os.listdir(keydir):
+        if x.endswith('.pub'):
+            with open(os.path.join(keydir, x)) as f:
+                key = json.load(f)['keyval']['public']
+                if pubkey == key:
+                    keyname = x[:-4]  # strip off .pub
+                    print("current root keyname:", keyname)
+                    return keyname, key_ids[0]
+    print('could not find root key name')
+    sys.exit(3)
+
+with TemporaryDirectory() as tempdir:
+    os.chdir(tempdir)
+    os.mkdir('tuf')
+    creds_file = '/tmp/creds.tgz'
+    try:
+      cmd('tar', 'xf', creds_file, cwd='./tuf')
+    except:
+      sys.exit(1)
+    try:
+      cmd('garage-sign', 'root', 'pull', '--repo', './tufrepo')
+    except:
+      sys.exit(2)
+    try:
+      old_keyname, old_keyid = find_current_root('./tuf/tufrepo')
+    except:
+      sys.exit(3)
+    keyname = 'offline-root-' + datetime.datetime.now().isoformat()
+    try:
+      cmd('garage-sign', 'key', 'generate', '--repo', './tufrepo',
+          '--type', 'rsa', '--name', keyname)
+    except:
+      sys.exit(4)
+    try:
+      cmd('garage-sign', 'root', 'key', 'add', '--repo', './tufrepo',
+          '--key-name', keyname)
+    except:
+      sys.exit(5)
+    try:
+      cmd('garage-sign', 'root', 'key', 'remove', '--repo', './tufrepo',
+          '--key-id', old_keyid, '--key-name', old_keyname)
+    except:
+      sys.exit(6)
+    try:
+      cmd('garage-sign', 'root', 'sign', '--repo', './tufrepo',
+          '--key-name', keyname, '--key-name', old_keyname)
+    except:
+      sys.exit(7)
+    try:
+      cmd('tar', 'czf', creds_file, 'tufrepo', cwd='./tuf')
+    except:
+      sys.exit(8)
+    try:
+      cmd('garage-sign', 'root', 'push', '--repo', './tufrepo')
+    except:
+      sys.exit(9)
+`
+	progress := newProgress("Pulling aktualizr image %s")
+	if err := pullContainer(aktualizrImageName); err != nil {
+		progress.fail()
+		exitf("failed to pull image, %q", aktualizrImageName)
```

Comment:
I just hit this: You aren't printing the error that occurs.

Reply:
good catch - fixed